s2a.go

```diff
@@ -279,7 +279,9 @@ func (c *s2aTransportCreds) OverrideServerName(serverNameOverride string) error
 }
 
 // TLSClientConfigOptions specifies parameters for creating client TLS config.
-type TLSClientConfigOptions struct{}
+type TLSClientConfigOptions struct {
+	ServerName string
```

Comment:
Can you clarify how we expect the user to populate this value in the HTTP case?

Reply:
here's an example, I also added it to the comment description for the field
		tlsConfig, err := factory.Build(ctx, &s2a.TLSClientConfigOptions{
			ServerName: "example.com",
		})

Reply:
Sorry let me clarify: in the GUAC HTTP code, how do you know what value you should populate the `ServerName` field with?